### PR TITLE
Add customerRef field and make code more consistent with other functi…

### PR DESCRIPTION
**Pull request by Soccerama**
…ons within abettor.

I've added the customerRed field which was missing from the Betfair operation. Have also made the internal code consistent with other functions and it now uses jsonlite rather than coding json directly.

Have done some testing and also update documentation. Would welcome any feedback if there are any issues.

### CODE REVIEW

File: R/cancelOrders.R [hunk 1]
```diff
@@ -97,27 +98,64 @@
...
-    cancelOrderOps = gsub("NULL","",cancelOrderOps)
+
+    show(cancelOrdersOps)
+
+    cancelOrdersOps <-
+      cancelOrdersOps[c("jsonrpc", "method", "params", "id")]
+
+    cancelOrdersOps <- jsonlite::toJSON(jsonlite::unbox(cancelOrdersOps))
+
+    show(cancelOrdersOps)
```
Comment (phillc73) [inline R/cancelOrders.R:158]: I don't think the show() is needed here and may be an artefact from your testing

File: R/cancelOrders.R [hunk 2]
```diff
@@ -97,27 +98,64 @@
 #'
 
 cancelOrders <-
-  function(marketId, betIds = NULL, sizeReductions = NULL, suppress = FALSE, sslVerify = TRUE) {
+  function(marketId = NULL, betIds = NULL, sizeReductions = NULL, customerRef = NULL, suppress = FALSE, sslVerify = TRUE) {
```
Comment (phillc73) [inline R/cancelOrders.R:101]: I think it's super dangerous here to have `marketId = NULL`. If the end user simply executes `cancelOrders()`, all bets across all markets are cancelled.

Back in the mists of time, I think I deliberately didn't set `marketID = NULL` from a risk perspective.

**Review by phillc73 — CHANGES_REQUESTED** [review]
Thanks for the really useful update. I guess old code like this sometimes lives on over the years.

Just two things I've commented about inline (let me know if you don't see the comments).

File: R/cancelOrders.R (hunk 2 above)
Comment (Soccerama) [reply R/cancelOrders.R:101]: Oops, yes, the show() shouldn't be there.

Good point on the risk around marketId = NULL. My thought here was that it is not a required parameter by the API but agree it is much safer to require the user to spell out that they want to cancel all orders.

**Comment by Soccerama** [comment]
I've made the changes you suggested. The logic was bit trickier because none of the parameters are required. The code is not elegant but it passed my testing.
There was also some duplication in my code which I removed. The code requires a marketId if betIds are supplied which I didn't think was necessary but it appears to be.

**Pull request merged**